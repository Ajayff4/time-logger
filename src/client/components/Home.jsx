import React, { Component } from 'react'
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom"
import Login from './Login'
import Logs from './Logs'
import AddTag from './AddTag'
import Profile from './Profile'
import AddLog from './AddLog'
import Signup from './Signup'
import AddCategory from './AddCategory'
import { logout } from '../actions'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'

class Home extends Component {
    // eslint-disable-next-line
    constructor(props) {
        super(props)
    }
    handleLogout = () => {
        this.props.logout()
    }
    render() {
        return (
            <nav id="menu-bar">
                <Router>
                    <ul>
                        <li key="login"><Link to="/login"><button>Login</button></Link></li>
                        <li key="add-log"><Link to="/add-log"><button>Add Log</button></Link></li>
                        <li key="logs"><Link to="/logs"><button>Logs</button></Link></li>
                        <li key="profile"><Link to="/profile"><button>Profile</button></Link></li>
                        <li key="signup"><Link to="/signup"><button>Signup</button></Link></li>
                        <li><button onClick={this.handleLogout}>Logout</button></li>
                    </ul>
                    {
                        <Switch>
                            <Route path="/login"><Login /></Route>
                            <Route path="/add-log"><AddLog /></Route>
                            <Route path="/logs"><Logs /></Route>
                            <Route path="/add-tag"><AddTag /></Route>
                            <Route path="/add-category"><AddCategory /></Route>
                            <Route path="/profile"><Profile /></Route>
                            <Route path="/signup"><Signup /></Route>
                        </Switch>
                    }
                </Router>
            </nav>
        )
    }
}

Home.propTypes = {
    logout: PropTypes.func.isRequired
}

const mapDispatchToProps = dispatch => {
    return {
        logout: () => dispatch(logout())
    }
}

export default connect(null, mapDispatchToProps)(Home)