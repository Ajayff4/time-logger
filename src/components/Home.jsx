import React from 'react'

export default function Home() {
    return (
        <center>
            <div className="homePageText">
            <h2 >This app is useful for managing time and also useful for self evaluation.</h2>
            <p>Made By AA077457</p>
            <p>Thank You!!!</p>
            </div>   
        </center>
    )
}