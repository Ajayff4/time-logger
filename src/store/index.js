import { createStore, applyMiddleware } from 'redux'
import rootReducer from '../reducers/index'
//import logger from 'redux-logger'
import { composeWithDevTools } from 'redux-devtools-extension'
import rootSaga from '../saga'
import createSagaMiddleware from 'redux-saga'

const sagaMiddleware = createSagaMiddleware()
const store = createStore(
    rootReducer,
    composeWithDevTools(
        applyMiddleware(sagaMiddleware))
)

sagaMiddleware.run(rootSaga)

export default store