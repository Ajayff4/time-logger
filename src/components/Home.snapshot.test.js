import React from 'react'
import renderer from 'react-test-renderer'
import Home from './Home'
import { Provider } from 'react-redux';
import store from '../store';
import { BrowserRouter as Router } from 'react-router-dom';

describe('Home Snapshot Testing...', () => {
    it('renders component correctly', () => {
        const tree = renderer.create(<Provider store={store}><Router><Home /></Router></Provider> ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});