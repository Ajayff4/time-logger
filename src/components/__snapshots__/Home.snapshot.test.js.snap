// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Snapshot Testing... renders component correctly 1`] = `
<div
  className="homeComponent"
>
  <center>
    <div
      className="homePageText"
    >
      <h2>
        This app is useful for managing time and also useful for self evaluation.
      </h2>
      <p>
        Made By AA077457
      </p>
      <p>
        Thank You!!!
      </p>
    </div>
  </center>
</div>
`;
